plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/delete/DummyFileScanTask.java

```diff
@@ -0,0 +1,231 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg.delete;
+
+import org.apache.iceberg.CombinedScanTask;
+import org.apache.iceberg.DataFile;
+import org.apache.iceberg.DataTask;
+import org.apache.iceberg.DeleteFile;
+import org.apache.iceberg.FileFormat;
+import org.apache.iceberg.FileScanTask;
+import org.apache.iceberg.PartitionSpec;
+import org.apache.iceberg.StructLike;
+import org.apache.iceberg.expressions.Expression;
+
+import java.nio.ByteBuffer;
+import java.util.List;
+import java.util.Map;
+
+import static java.util.Objects.requireNonNull;
+
+// TODO: This wrapper is necessary until the constructors of the Iceberg DeleteFilter are made more specific
+// Remove after merging https://github.com/apache/iceberg/pull/4381
```

Comment:
```suggestion
// Remove after upgrading to iceberg with https://github.com/apache/iceberg/pull/4381
```

Reply:
Can we have eg reflection-based test to ensure the TODO gets done?

Reply:
This line will stop compiling after pulling in the updated Iceberg library with that PR https://github.com/trinodb/trino/pull/11642/files#diff-bea7e01e5f9c44ac82298bfe3334cc2db359a751723913112135494273025cd8R40